sdk/eventhubs/azure-messaging-eventhubs/migration-guide.md

```diff
@@ -0,0 +1,353 @@
+* [Receiving events with checkpointing](#migrating-code-from-eventprocessorhost-to-eventhubconsumerclient-for-receiving-events)
+
+### Migrating from `PartitionSender` or `EventHubClient` to `EventHubProducerAsyncClient` for sending events
+In v3, there were multiple options on how to publish events to an Event Hub.
+
+In v5, this has been consolidated into a more efficient `send(EventDataBatch)` method. Batching merges information from
+multiple events into a single sent message, reducing the amount of network communication needed vs sending events one at
+a time.
+
+The code below assumes all events fit into a single batch. For a more complete example, see sample: [Publishing events
+to specific partition][PublishEventsToSpecificPartition].
+
+So in v3:
+```java
+EventHubClient client = EventHubClient.createFromConnectionString("connection-string-for-an-event-hub",
+    Executors.newScheduledThreadPool(4)).get();
+
+List<EventData> events = Arrays.asList(EventData.create("foo".getBytes()), EventData.create("bar".getBytes()));
+
+CompletableFuture<Void> sendFuture = client.createPartitionSender("my-partition-id")
+    .thenCompose(sender -> {
+        EventDataBatch batch = sender.createBatch();
+        for (EventData event : events) {
+            try {
+                // Assuming all events fit into a single batch. This returns false if it does not.
+                // If that is the case, we'll send the full batch then create another one to continue adding events to.
+                batch.tryAdd(event);
+            } catch (PayloadSizeExceededException e) {
+                System.err.println("Event is too large for batch. Exception: " + e);
+            }
+        }
+
+        return sender.send(batch);
+    });
+sendFuture.get();
+```
+
+In v5:
+```java
+List<EventData> events = Arrays.asList(EventData.create("foo".getBytes()), EventData.create("bar".getBytes()));
+
+EventHubProducerAsyncClient producer = new EventHubClientBuilder()
+    .connectionString("connection-string-for-an-event-hub")
+    .buildAsyncProducerClient();
+
+CreateBatchOptions options = new CreateBatchOptions()
+    .setPartitionId("my-partition-id");
+
+Mono<Void> sendOperation = producer.createBatch(options).flatMap(batch -> {
+    for (EventData event : data) {
+        // Assuming all events fit into a single batch. This returns false if it does not.
+        // If that is the case, we'll send the full batch then create another one to continue adding events to.
+        batch.tryAdd(event);
+    }
+    return producer.send(batch);
+});
+
+sendOperation.block();
+```
+
+### Migrating code from `PartitionReceiver` to `EventHubConsumerClient` for receiving events
+
+In V2, event handlers were passed as positional arguments to `receive`.
+
+In V5, event handlers are passed as part of a `SubscriptionEventHandlers` shaped object.
+
+For example, this code which receives from a partition in V2:
+
+```typescript
```

Comment:
nit:  I think these are meant to be `java`, no?   Same for line 183.

Reply:
Maybe not.....   but that's a bit weird.  Is this section correct?

Reply:
Now it makes sense, I forgot to actually push my latest version.. so some of these comments didn't add up. LOL. 🤣
Connie: "Ctrl+F. Where is this code snippet?!!"